### ISSUE TITLE
[Bug] Give a summary of the problem

### ISSUE THREAD

**Issue by Alberth289346** [CorsixTH/CorsixTH#2177]
### Describe the issue

Handyman window does not show the handyman properly while busy.
![Screenshot from 2022-05-29 16-41-30](https://user-images.githubusercontent.com/3254232/170882392-1476eb19-8554-4f1f-a2c8-2033aad40bc2.png)
![Screenshot from 2022-05-29 16-40-25](https://user-images.githubusercontent.com/3254232/170882416-e2fc5d4a-5f22-4229-9f96-888c030b800c.png)

As you can see, the handyman is not centered in the circular area in his window.

Bug doesn't seem new, both shots are from a development version, but I have confirmed the sweeping screenshot also in fd7a9420 (2 fixes further than `v0.66-beta1`) and in `v0.65`.

### Steps to Reproduce

Likely any not-too-old save will work, but from scratch, build a minimal hospital, and wait a while.

### Expected Behaviour

Handyman should be standing properly in front of the camera of course!

### Save Game

Save game attached to report: Yes/No

### CorsixTH Version

0.65 to 0.66-beta1

### Operating System

Mint LTS

### Theme Hospital Version

CD

### Gamelog.txt

```bash
nothing of interest.
```

**Comment by lewri** [comment]
It looks like the mini window doesn't know where to focus as the handyman repairs... Unless when repairing it makes the game think he is standing on a specific tile in respect of the machine?

**Comment by Alberth289346** [comment]
I am pretty sure the handyman is standing on the tile being sweeped (liked that better than swept :) ). I checked the sweep animation (1878), and the handy man is standing at tile (0,0):
![image](https://user-images.githubusercontent.com/3254232/171044890-107cd0a2-cc2a-4a9d-98a1-f90e88395a43.png)

It also has a `setMarker` call, and for fun I changed that:
``` diff
diff --git a/CorsixTH/Lua/humanoid_actions/sweep_floor.lua b/CorsixTH/Lua/humanoid_actions/sweep_floor.lua
index bc4a9d51..bcab4655 100644
--- a/CorsixTH/Lua/humanoid_actions/sweep_floor.lua
+++ b/CorsixTH/Lua/humanoid_actions/sweep_floor.lua
@@ -35,7 +35,7 @@ local animation_numbers = {
   1874,
   1878,
 }
-TheApp.animation_manager:setMarker(animation_numbers, {-1.05, -0.05})
+TheApp.animation_manager:setMarker(animation_numbers, {-0.05, -0.05})
 
 local finish = permanent"action_sweep_floor_finish"( function(humanoid)
   humanoid:finishAction()

```
This seems to fix (or at least improve) the offset while sweeping. That line hasn't changed since 2012, so maybe something else changed?

**Comment by lewri** [comment]
I remember reading somewhere that when a humanoid has an object animation, they become part of the object I think. That might be why they disappear? And likely yes, the offset is wrong. You may have to actually define the offset for each object use and orientation?

**Comment by lewri** [comment]
This bit of CPP reports back the marker of the humanoid for the mini-window. May be useful:
https://github.com/CorsixTH/CorsixTH/blob/98186d60f56a2871d38d4a2689859e39d138bc55/CorsixTH/Src/th_lua_anims.cpp#L562-L570

**Comment by Alberth289346** [comment]
> I remember reading somewhere that when a humanoid has an object animation, they become part of the object I think. That might be why they disappear? And likely yes, the offset is wrong. You may have to actually define the offset for each object use and orientation?

Correct. the humanoid hides itself by disabling its animation while the object animation is running, and re-appears afterwards.
We have these marker things that basically track the head position for the mood icon (and apparently for the mini-window too).

I highly doubt it's the right solution though. If I had an animation system with layers to add clothes and shoes to a humanoid, I would add a layer for tracking the head position rather than invent an additional system where you manually have to figure out the position of the head in the frames of the animation.

**Comment by lewri** [comment]
@Alberth289346 I think #2824 fixes the two you screenshotted. Likely I'll look at fixing the others slowly.

**Comment by Alberth289346** [comment]
Yep
